### ISSUE TITLE
Makefile instructions?

### ISSUE THREAD

**Issue by smiths** [caobo1994/FourierSeries#37]
@Ao99, does your code include any instructions for modifying the makefile for local installation?  Your code depends on several include files, like eigen3, boost, and on various libraries.  Your users will need to know what they need to install, and how they need to modify your makefile.

I modified the file so that the includes worked and the object files were generated, but the linker failed.  I pointed it to the gcc folder on my Mac (by modifying your -L directive), but that was not adequate.  I'm not going to invest any more time in linking.  I can grade your project without running the code.  However, if you want people to use your code, you will need to make it easier for them to know the dependencies and modify the makefile.  The explanation typically goes in the INSTALL.txt file.

I also tried running your code on mills, but I think the boost library on mills is older than what you require.

**Comment by caobo1994** [comment]
Sorry for the trouble. I should provide the clang version as well. 

As to the third-party libraries, the common practice in C/C++ libraries is to specify the used libraries and required versions, and let the users deploy these libraries and specify their locations.

The content about the dependant libraries shall be in the user guide file, and I will include them.

**Comment by caobo1994** [comment]
The current makefile is for testing only. The deployment of this library only includes copying the source file to the location chosen by the user. (which is the common practice of distributing header-only files, and this is the knowledge that second-year CS/SE students, who have finished their C and C++ courses, should know.)

**Comment by smiths** [comment]
Yes, I was looking for a list of required libraries and version numbers.  It also helps to include comments in the include file to let users know the parts they may need to change.

**Comment by caobo1994** [comment]
@smiths, UserGuide updated in commit a7ecc80.

**Comment by caobo1994** [comment]
@smiths 

Clang version amended in commit e6b3cab.

Anyone who would like to use it should use `r_clang` script to replace `r`.
